### PR TITLE
[simd] Implement load lane

### PR DESCRIPTION
**Pull request by ngzhian**
This is a new kind of ir/ast node/instruction. It has 3 immediates:
memarg align, memarg offset, and lane index. This required new visitor
functions in all the places.

Drive-by cleanup to share the simd lane parsing logic between shuffle,
lane op and this new load lane instructions. This requires rebasing some
tests because the error messages are slightly different now.

### CODE REVIEW

File: src/binary-reader.cc [hunk 1]
```diff
@@ -1318,6 +1318,21 @@ Result BinaryReader::ReadFunctionBody(Offset end_offset) {
...
+      case Opcode::V128Load64Lane: {
+        Address alignment_log2;
+        CHECK_RESULT(ReadAlignment(&alignment_log2, "load alignment"));
+        Address offset;
+        CHECK_RESULT(ReadAddress(&offset, 0, "load offset"));
+        uint8_t lane_val;
+        CHECK_RESULT(ReadU8(&lane_val, "Lane idx"));
+
+        CALLBACK(OnSimdLoadLaneExpr, opcode, alignment_log2, offset, lane_val);
+        CALLBACK(OnOpcodeUint32Uint32, alignment_log2, offset);
```
Comment (sbc100) [inline src/binary-reader.cc:1333]: Shouldn't this be `OnOpcodeUint32Uint32Uint32`  and include `lave_val` ?   (I guess that doesn't exist yet?)

Comment (ngzhian) [reply src/binary-reader.cc]: Yea you're right, added and fixed.

**Comment by ngzhian** [comment]
@sbc100 ptal, thanks!

**Review by sbc100 — COMMENTED** [review]
Should there be an objdump change here?  Should we have test that we can disassembly/dump this new instruction?

**Comment by ngzhian** [comment]
Good idea, found a bug in the dumping, wasn't overriding the new uint32uint32uint32 correctly. Added the test. Ptal, thanks!

**Comment by ngzhian** [comment]
@sbc100 ptal, thanks!

**Review by sbc100 — APPROVED** [review]

**Pull request merged**